git/git_branches_intro.md

```diff
@@ -1,22 +1,48 @@
 # Git Branches Introduction
-A brief summary about what this lesson is about and why the topics or concepts it covers are important.
+Git branches are a fundamental piece of the Git framework. This lesson will teach you how to think conceptually about them and know what they are useful for, as well as why. By the end you should be able to create your own branch and use it to contribute!
 
-## Learning outcomes
-*Look through these now and then use them to test yourself after doing the assignment*
+## Learning Outcome
+*By the end of this lesson you should be able to:*
 
-what the student is expected to know or be able to do by the end of this lesson
 
-* Learning outcome 1
-* Learning outcome 2
-* Learning outcome 3
+* Know what a Git branch is.
+* Understand what the master branch is.
+* Understand why branches are useful in projects, especially in a team environment.
+* Understand feature branches and their purpose.
+* Understand what a spike branch is for.
 
 ## Assignment
-The list of resources the user will go through to learn about the topic of this lesson. Have no more than 5 resources, ideally no more than three.
+* [Read this tutorial on Git branches](https://www.atlassian.com/git/tutorials/using-branches). This will cover some commands, and information on what branches are.
+* [Checkout this video](https://www.youtube.com/watch?v=JTE2Fn_sCZs) to see git branching in action, as well as to learn how feature branches are useful on development teams.
+
+## Branch Types
+To fully understand branches, you will want to learn and use the different types.
+* **Master** : The name commonly given to the main branch, where everyone on a development team will push their changes to.
+* **Feature** : A branch created for an individual feature to be worked on. These can be divided amongst team members without fear of interferring with eachothers code.
+* **Develop** : A branch used to actively develop on.
+* **Release or Spike** : This is a branch that is not developed on, but is instead a snapshot, usually ready for release. Think of them as "Spikes" in the ground, or immovable.
```

Comment:
I've always known spike branches as branches which you are doing an experiment in,  it is a branch you may continue working in if the experiment goes well or you may abandon it if you determine what you are trying to do isn't worth it/feasible.

I appreciate spike branches might have different meanings based on different peoples perspectives though.